### PR TITLE
docs: improve visualization chapter

### PR DESCRIPTION
**Pull request by takam5f2**

### CODE REVIEW

**Review by keita1523 — COMMENTED** [review]
minor suggestion

File: docs/visualization/filter/lttng_event_filter.md [hunk 1]
```diff
@@ -1,39 +1,52 @@
...
-- duration_filter
-- strip_filter
+- `init_pass_filter`
+- `duration_filter`
+- `strip_filter`
 
-## API Description
+## `init_pass_filter`
+
+With `init_pass_filter` method, events recorded by initialization tracepoints are not loaded onto memory.
```
Comment (hsgwa) [inline docs/visualization/filter/lttng_event_filter.md:15]:
```suggestion
With `init_pass_filter` method, events recorded by runtime tracepoints are not loaded onto memory.
```

File: docs/visualization/index.md [hunk 2]
```diff
@@ -1,41 +1,73 @@
...
+### Communication
 
-- [create_communication_latency_plot](./communication/communication_information.md#latency)
-- [create_communication_frequency_plot](./communication/communication_information.md#frequency)
-- [create_communication_period_plot](./communication/communication_information.md#period)
-- [create_publish_subscription_period_plot](./communication/pub_sub_information.md#period)
-- [create_publish_subscription_frequency_plot](./communication/pub_sub_information.md#frequency)
+- [`create_communication_latency_plot`](./communication/index.md#latency)
+- [`create_communication_frequency_plot`](./communication/index.md#frequency)
+- [`create_communication_period_plot](./communication/index.md#period)
```
Comment (hsgwa) [inline docs/visualization/index.md:59]:
```suggestion
- [`create_communication_period_plot`](./communication/index.md#period)
```

**Review by hsgwa — CHANGES_REQUESTED** [review]

File: docs/visualization/callback/index.md [hunk 3]
```diff
@@ -73,11 +76,13 @@ plot.show()
 
 ![callback_period_time_line](../../imgs/callback_period_time_line.png)
 
-The horizontal axis means `Time [s]` (changeable in ['system_time', 'sim_time', 'index']).
-The vertical axis means `Period [ms]` from one callback starts to the next time that the callback starts.
+The horizontal axis means time, labeled as `Time [s]`. `xaxis_type` argument is prepared as well as `create_callback_frequency_plot()` method.
+The vertical axis means period of callback execution, labeled as `Period [ms]`. It is plotted per sample.
```
[the remark this answers is not in the document]
Comment (takam5f2) [reply docs/visualization/callback/index.md]: Fixed at https://github.com/tier4/CARET_doc/pull/119/commits/1d45655d8182a3bb9e13e99cc40340ddf5a32f24

File: docs/visualization/communication/publish_subscription.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+<prettier-ignore-start>
+!!!info
+    1. The `output_notebook()` is needed to show figures inside Jupyter Notebook.
+    2. The function `get_publishers()` and `get_subscriptions()` return a List of publishers/subscribers involving the given topic name. The sample code uses the first value of the List.
+<prettier-ignore-end>
+
+## Frequency
+
+`Plot.create_publish_subscription_frequency_plot(Collection[publish: Publisher or subscription: Subscriber])` is introduced to check whether target publisher or frequency.
```
[the remark this answers is not in the document]
Comment (takam5f2) [reply docs/visualization/communication/publish_subscription.md]: Fixed at https://github.com/tier4/CARET_doc/pull/119/commits/08af126e4cda672684d6cee88aecc0075ee6a66c

File: docs/visualization/communication/publish_subscription.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
+# Publish and Subscription
+
+CARET can visualize frequency and invocation period per publish or subscription.
+This section describes sample visualization scripts for them.
+Execute the following script code to load trace data and an architecture object before calling visualization API.
+
+This visualization show you when publish or subscription is invoked in a target application.
```
[the remark this answers is not in the document]
Comment (takam5f2) [reply docs/visualization/communication/publish_subscription.md]: Fixed at https://github.com/tier4/CARET_doc/pull/119/commits/08af126e4cda672684d6cee88aecc0075ee6a66c

File: docs/visualization/callback/index.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (takam5f2) [reply docs/visualization/callback/index.md]: @keita1523 
I'd like to resolve this comment. Please feel free to reopen this if you have an additional comment.

File: docs/visualization/communication/publish_subscription.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (takam5f2) [reply docs/visualization/communication/publish_subscription.md]: @keita1523
I'd like to resolve this comment. Please feel free to reopen this if you have an additional comment.

File: docs/visualization/communication/publish_subscription.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (takam5f2) [reply docs/visualization/communication/publish_subscription.md]: @keita1523
I'd like to resolve this comment. Please feel free to reopen this if you have an additional comment.

File: docs/visualization/communication/index.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+<prettier-ignore-start>
+!!!info
+    This premise section may be boring for some users. If you don't have much time, please feel free to skip the section.
+<prettier-ignore-end>
+
+In ROS 2, nodes communicate each other via topic messages. The topic messages are transmitted by publisher and received by subscription. The following figure shows the simplest example where one node sends topic message and the other node receives it.
+
+![simple communication](./imgs/simple_communication.svg)
+
+CARET serves `Communication` class which focus on a topic message from a source node to a destination node. A `Communication`-based object is retrieved from method, `Application.get_communication('source node', 'destination node', '/topic/message')`. A `Communication` object has a collection of timestamps which are obtained when both publish and subscription on a target message are performed successfully. `Communication` class is explained in the page.
```
[the remark this answers is not in the document]
Comment (takam5f2) [reply docs/visualization/communication/index.md]: @hsgwa  
I understand what you mean. Does it seem good if I add `info` or `note` to tell that our intention is different from the current implementation?

File: docs/visualization/communication/many_to_many_communication.md [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+## Caution for Many-to-1 communication
+
+CARET requires users to take care of Many-to-1 communication. Many-to-1 communication means that topic messages are published from multiple nodes and received by a single node as the following figure shows.
+
+![multiple publisher](./imgs/multiple_publisher.svg)
+
+In this case, invocation frequency of publish on the source node is different from that of subscription on the destination node. The destination node receives topic messages from 2 other nodes. It is expected that sum of publish frequency on three nodes is equaled to that of subscription.
+
+If you see that publish frequency is different from subscription frequency, you may think loss of topic messages. However, it is reasonable when many-to-1 communication is performed.
```
[the remark this answers is not in the document]
Comment (takam5f2) [reply docs/visualization/communication/many_to_many_communication.md:18]: Added more explanation at https://github.com/tier4/CARET_doc/pull/119/commits/4a532050fc7c1109518d77b88c1b21c35b4f301c

**Comment by takam5f2** [comment]
@hsgwa @keita1523  
Thank you for reviewing this PR. I'd like to close this because this PR is carried to #131

**Pull request closed without merging**